hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -138,6 +138,7 @@ case class TypeParserEnvironment(
 }
 
 case class IRParserEnvironment(
+  ctx: ExecuteContext,
```

Comment:
IIUC, you only need the `FS` from this, right? Seems weird to put something called "ExecuteContext" into the parser environment. This is Tim's wheelhouse though, so I don't fully understand the intentions of IRParserEnvironment.

Reply:
Yeah, it is weird.  Putting an FS in it also seems weird, but that's the reality of having an IR node with a path whose type is determined by contents of the file at that path.  I can pull out just the FS.  The reference genomes will also move to the execute context and is needed by the parser, so this seems appropriate, but happy to break them out individual if you prefer.